packages/@react-aria/selection/src/useSelectableItem.ts

```diff
@@ -143,6 +143,13 @@ export function useSelectableItem(options: SelectableItemOptions): SelectableIte
       } else {
         focusSafely(ref.current);
       }
+
+      // TODO: is this the appropriate place to put this? We need the focused DOM node
+      // Also keyboard modality != keyboard navigation. If the manager.focusedKey changes programatically and not
+      // in response to keypress but the last interaction modality was keyboard then we'd perform the scroll
+      if (getInteractionModality() === 'keyboard') {
+        scrollIntoViewFully(ref.current);
+      }
```

Comment:
Still a bit wary about this, digging to see if there is anywhere in the code that actually will update the focusedKey in response to a non-keyboard input + when in keyboard modality